subsys/net/lib/lwm2m/ipso_pressure_sensor.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2018-2019 Foundries.io
+ * Copyright (c) 2020 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Source material for IPSO Sensor object (3303):
+ * https://github.com/IPSO-Alliance/pub/blob/master/docs/IPSO-Smart-Objects.pdf
+ */
+
+#define LOG_MODULE_NAME net_ipso_pressure_sensor
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <stdint.h>
+#include <init.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_resource_ids.h"
+
+#ifdef CONFIG_LWM2M_IPSO_PRESSURE_SENSOR_TIMESTAMP
+#define ADD_TIMESTAMPS 1
+#define NUMBER_OF_OBJ_FIELDS 8
+#else
+#define ADD_TIMESTAMPS 0
+#define NUMBER_OF_OBJ_FIELDS 7
+#endif
+
+#define MAX_INSTANCE_COUNT CONFIG_LWM2M_IPSO_PRESSURE_SENSOR_INSTANCE_COUNT
+
+#define IPSO_OBJECT_ID PRESSURE_OBJECT_RID
+
+#define SENSOR_NAME "Pressure"
+
+#define UNIT_STR_MAX_SIZE 8
+
+/*
+ * Calculate resource instances as follows:
+ * start with NUMBER_OF_OBJ_FIELDS
+ * subtract EXEC resources (1)
+ */
+#define RESOURCE_INSTANCE_COUNT (NUMBER_OF_OBJ_FIELDS - 1)
+
+/* resource state variables */
+static float32_value_t sensor_value[MAX_INSTANCE_COUNT];
+static char units[MAX_INSTANCE_COUNT][UNIT_STR_MAX_SIZE];
+static float32_value_t min_measured_value[MAX_INSTANCE_COUNT];
+static float32_value_t max_measured_value[MAX_INSTANCE_COUNT];
+static float32_value_t min_range_value[MAX_INSTANCE_COUNT];
+static float32_value_t max_range_value[MAX_INSTANCE_COUNT];
+
+static struct lwm2m_engine_obj sensor;
+static struct lwm2m_engine_obj_field fields[] = {
+	OBJ_FIELD_DATA(SENSOR_VALUE_RID, R, FLOAT32),
+	OBJ_FIELD_DATA(UNITS_RID, R_OPT, STRING),
+	OBJ_FIELD_DATA(MIN_MEASURED_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_DATA(MAX_MEASURED_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_DATA(MIN_RANGE_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_DATA(MAX_RANGE_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_EXECUTE_OPT(RESET_MIN_MAX_MEASURED_VALUES_RID),
```

Comment:
What about optional "Current Calibration" (5821) and "Application Type" (5750) resources? They're specified in
https://raw.githubusercontent.com/OpenMobileAlliance/lwm2m-registry/prod/3323.xml

Reply:
Following the implementation of temp_sensor, so not implementing these in the pressure sensor at this time.